### PR TITLE
#136 filter function

### LINKED ISSUE #136: Implement filter function in browser
Verify filter, including search function is working correctly and fix the css to match our style.

### PR DESCRIPTION
**Pull request by jakobemilandersson**
#### What does this PR do?
Integrates the filter-function on the client with the back-end and makes the filter look more like the decided design.
(Please tell me if something should be changed with the current design!)
[Issue](LINK_TO_ISSUE)
#136 
##### Why are we doing this? Any context or related work?
To make it possible for the user to filter through all the bike ads.
#### Where should a reviewer start?
Filter.js
#### Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).
closes #136

### CODE REVIEW

File: Bikeini/components/Filter.js [hunk 1]
```diff
@@ -166,18 +188,31 @@ class Filter extends React.Component {
...
-    this.setState({ checkBoxes });
+    changeItems(checkBoxes);
   }
 
   renderRow(rowData) {
     return (
-      <View style={styles.rowContainer}>
+      <View style={styles.rowContainer} key={Math.random() * 10000}>
         <View style={styles.itemContainer}>
           <ItemCheckbox title={rowData[0].title} category={rowData[0].category} id={rowData[0].id} isChecked={rowData[0].isChecked} onChange={this.updateCheckBoxes} />
```
Comment (FredSoderberg) [inline Bikeini/components/Filter.js:217]: Why is there 3 empty "places"?

File: Bikeini/components/Filter.js [hunk 2]
```diff
@@ -189,35 +224,69 @@ class Filter extends React.Component {
...
-          }}
-          >
+        <KeyboardAwareScrollView
+          contentContainerStyle={styles.container}
+          enableOnAndroid
+          extraScrollHeight={100}
+          enableAutoAutomaticScroll={(Platform.OS === 'ios')}
+          innerRef={(ref) => { this.scroll = ref; }}
+          keyboardShouldPersistTaps="handled"
+        >
```
Comment (FredSoderberg) [inline Bikeini/components/Filter.js:238]: really Nice addition... but on iPhone XS and galaxy 6 its really yanky. Still better then nothing but I would guess there is a better package for this *host host*  👍

**Review by FredSoderberg — CHANGES_REQUESTED** [review]
Nice addition, just check one of the comments about missing fields.... and ESlint next time :P

File: Bikeini/components/Filter.js (hunk 1 above)
Comment (FredSoderberg) [reply Bikeini/components/Filter.js:217]: image

File: Bikeini/components/Filter.js (hunk 2 above)
Comment (FredSoderberg) [reply Bikeini/components/Filter.js:238]: Its a neat feature which needs to be included in more places 👍

File: Bikeini/components/Filter.js (hunk 1 above)
Comment (jakobemilandersson) [reply Bikeini/components/Filter.js:217]: To make it look like the design on Marvel! -> Seperating the two different "groups" of options. I can merge them togheter into one group instead. 😄

File: Bikeini/components/Filter.js (hunk 2 above)
Comment (jakobemilandersson) [reply Bikeini/components/Filter.js:238]: Yeah sadly it doesn't really work as it should on Android devices when running through Expo so it looks kind of weird... 😞 Will be looking for a better option but can remove it for this PR if it's too wonky!

File: Bikeini/components/Filter.js (hunk 1 above)
Comment (FredSoderberg) [reply Bikeini/components/Filter.js:217]: Ahhh sorry, misses That in the design!

File: Bikeini/components/Filter.js (hunk 2 above)
Comment (FredSoderberg) [reply Bikeini/components/Filter.js:238]: No let’s keep it, it fulfills it’s purpose almost ;) so it’s better than the alternative, we can write it up as an enhancement.

**Review by FredSoderberg — APPROVED** [review]
Really nice!

**Pull request merged**